File: frontend/src/types/order.ts
export type OrderStatus = "payment_pending" | "payment_failed" | "paid" | "confirmed" | "shipped" | "delivered" | "cancelled";

export interface OrderItem {
  id: number;
  product_id: number;
  product_name: string | null;
  quantity: number;
  unit_price: number;
}

export interface Order {
  id: number;
  user_id: number;
  user_email?: string | null;
  status: OrderStatus;
  total_amount: number;
  created_at: string;
  updated_at?: string;
  payment_status?: string | null;
  branch_id?: number | null;
  items?: OrderItem[];
}

export interface StatusHistoryEntry {
  id: number;
  estado_desde: string | null;
  estado_hasta: string;
  motivo: string | null;
  created_at: string;
}

export interface OrderDetail extends Order {
  payment_status: string | null;
  payment_method?: string | null;
  paid_at?: string | null;
  status_history?: StatusHistoryEntry[];
  items: OrderItem[];
}

export interface OrderListResponse {
  items: Order[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

File: backend/app/schemas/order.py
"""Order Pydantic schemas for validation and serialization."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class OrderItemResponse(BaseModel):
    """Order item response schema."""

    id: int
    order_id: int
    product_id: int
    product_name: str | None = None
    quantity: int
    unit_price: Decimal

    model_config = ConfigDict(from_attributes=True)


class OrderResponse(BaseModel):
    """Order list item response schema."""

    id: int
    user_id: int
    status: str
    total_amount: Decimal
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OrderDetailResponse(BaseModel):
    """Order detail response schema with items."""

    id: int
    user_id: int
    status: str
    total_amount: Decimal
    status_history: list[dict] | None = None
    payment_status: str | None = None
    payment_method: str | None = None
    paid_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemResponse] = []

    model_config = ConfigDict(from_attributes=True)


class OrderListResponse(BaseModel):
    """Paginated order list response."""

    items: list[OrderResponse]
    total: int
    page: int
    limit: int
    total_pages: int


class OrderStatusUpdate(BaseModel):
    """Schema for updating order status (admin)."""

    status: str = Field(..., min_length=1, max_length=20)


class HistorialResponse(BaseModel):
    """Order status history entry response."""

    id: int
    estado_desde: str | None = None
    estado_hasta: str
    motivo: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
